circuit_tutorials/plonky2/merkle_tree/src/main.rs

```diff
@@ -0,0 +1,238 @@
+            println!("Proving failed.");
+            std::process::exit(1);
+        }
+
+        println!("Saving proof details locally");
+        let file = File::create("./data/prove_out.json").unwrap();
+        let mut writer = BufWriter::new(file);
+        serde_json::to_writer_pretty(&mut writer, &proof_data).unwrap();
+        writer.flush().unwrap();
+    }
+}
+
+fn verify_proof(input: &str) {
+    let mut file = File::open(input).unwrap();
+    let mut contents = String::new();
+    file.read_to_string(&mut contents).unwrap();
+    let proof_details: Value = serde_json::from_str(&contents).unwrap();
+
+    let proof_data: JsonProofData = serde_json::from_value(proof_details["proof"].clone()).unwrap();
+
+    let proof_bytes = general_purpose::STANDARD.decode(proof_data.proof).unwrap();
+    let common_bytes = general_purpose::STANDARD.decode(proof_data.common).unwrap();
+    let verifier_only_bytes = general_purpose::STANDARD
+        .decode(proof_data.verifier_data)
+        .unwrap();
+
+    let default_gate_serializer = DefaultGateSerializer;
+
+    let common =
+        CommonCircuitData::<F, D>::from_bytes(common_bytes, &default_gate_serializer).unwrap();
+    let proof = ProofWithPublicInputs::<F, C, D>::from_bytes(proof_bytes, &common).unwrap();
+    let verifier_data = VerifierOnlyCircuitData::<C, D>::from_bytes(verifier_only_bytes).unwrap();
+
+    let verifier: VerifierCircuitData<F, C, D> = VerifierCircuitData {
+        verifier_only: verifier_data,
+        common: common,
+    };
+
+    match verifier.verify(proof) {
+        Ok(_) => println!("Proof has been verified!"),
+        Err(e) => println!("Verification failed with error: {:?}", e),
+    }
+}
+
+fn headers_json(api_key: &str) -> HeaderMap {
+    let mut headers_json = HeaderMap::new();
+    headers_json.insert("Accept", "application/json".parse().unwrap());
+    headers_json.insert(
+        "Authorization",
+        HeaderValue::from_str(&format!("Bearer {api_key}").to_string()).unwrap(),
+    );
+    headers_json
+}
+
+async fn poll_status(endpoint: String, api_url: &str, api_key: &str, timeout: i64) -> Value {
+    let client = Client::new();
+    for _i in 1..timeout {
+        let response = client
+            .get(format!("{api_url}{endpoint}"))
+            .headers(headers_json(api_key))
+            .send()
+            .await
+            .unwrap();
+        assert_eq!(
+            &response.status().as_u16(),
+            &200u16,
+            "Expected status code 201"
+        );
+
+        let data = response.json::<Value>().await.unwrap();
+        let status = &data["status"].to_string();
+        if ["Ready", "Failed"]
+            .iter()
+            .any(|&s| status.as_str().contains(s))
+        {
+            return data;
+        }
+
+        tokio::time::sleep(Duration::from_secs(1)).await;
+    }
+
+    println!("Polling timed out after {} seconds", timeout);
+    std::process::exit(1);
+}
```

Comment:
The functions in this file are examples of how to use Sindri API, so let's give each a nice docstring explaining what they are doing.

Reply:
Added docstrings to all functions: [deb894b](https://github.com/Sindri-Labs/sindri-resources/pull/98/commits/deb894b5c71501a11a263ca7b61b61050377ecb0)

Reply:
nice! I added a few inline suggestions since I agree with your comment about omitting details about where we store the json result locally.